lib/internal/errors.js

```diff
@@ -186,14 +193,16 @@ function lazyBuffer() {
   return buffer;
 }
 
-const addCodeToName = hideStackFrames(function addCodeToName(err, name, code) {
+const addCodeToName = hideStackFrames(function addCodeToName(err, name, code,
+                                                             deferStack) {
   // Set the stack
   err = captureLargerStackTrace(err);
   // Add the error code to the name to include it in the stack trace.
   err.name = `${name} [${code}]`;
```

Comment:
if we moved this logic into `prepareStackTrace` we could get rid of the `err.stack` call in all cases, avoid the extra map, and save some perf.

Reply:
The problem is the `delete err.name` a few lines below - not sure how to handle the state reversion consistently.

Reply:
it is being set for Error.prototype.toString to read it, so you can just modify the area where Error.prototype.toString is called to include the code, instead of modifying the name property

Reply:
The thing is we want to ensure we only do this mutation for Node.js errors and not user errors, otherwise we will reformat eg all errors which have a code.

I'm quite surprised there isn't a standard branding system / custom formatting system and agree that's a better architecture, but asking me to implement it right now seems a little tall Gus.

Reply:
i don't think it would be that complex:

```js
if (error instanceof BaseError) {
  output = `${error.name} [${error.code}]: ${error.message}`
} else {
  output = ErrorPrototypeToString(error);
}
```

Reply:
Unfortunately there isn't a standard Node.js `BaseError` - all errors are based off of standard JS error classes without extension, only a possible multiple inheritance of multiple error classes.

Reply:
We could add a `kNodeJsError` symbol property to Node.js error objects.

Reply:
i'll open a pr.

Reply:
Yes I suppose that could work for such a refactoring, but I don't see why that is in scope of this PR since it is an existing architecture.

Reply:
Thanks for the PR on this - I've rebased to that now.